File: ANGEN/SampleFramework.Controls/SmoothControls - User's Guide/Samples/Toolbars Category/Command Manager/CS/CommandManager/MainForm.cs
using Genetibase.Shared.Controls;

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace CommandManager
{
	public partial class MainForm : Form
	{
		private void _commandManager_ApplicationCommandUpdate(object sender, NuGenApplicationCommandEventArgs e)
		{
			Control owner = sender as Control;
			NuGenApplicationCommand command = e.ApplicationCommand;
			ToolStripItem item = e.Item as ToolStripItem;

			switch (command.ApplicationCommandName)
			{
				case "New":
				{
					e.ApplicationCommand.Enabled = _enableNewCheckBox.Checked;
					break;
				}
				case "FileCopy":
				{
					e.ApplicationCommand.Enabled = _enableGlobalCopyCheckBox.Checked;
					break;
				}
				case "ContextCopy":
				{
					e.ApplicationCommand.Enabled = _enableLocalCopyCheckBox.Checked;
					break;
				}
				case "Save":
				{
					e.ApplicationCommand.Visible = _showSaveCheckBox.Checked;
					break;
				}
				case "TrackBar":
				{
					_progressBar.Value = _trackBar.Value;
					break;
				}
			}
		}

		public MainForm()
		{
			this.InitializeComponent();
			this.SetStyle(ControlStyles.Opaque, true);
		}
	}
}

File: ANGEN/SampleFramework.Controls/MiniBar/Samples/Toolbars Category/Command Manager/VB/CommandManager/MainForm.vb
Imports Genetibase.Shared.Controls
Imports System.Windows.Forms

Public Class MainForm
    Private Sub _commandManager_ApplicationCommandUpdate(ByVal sender As System.Object, ByVal e As NuGenApplicationCommandEventArgs) Handles _commandManager.ApplicationCommandUpdate
        Dim owner As Control = CType(sender, Control)
        Dim Command As NuGenApplicationCommand = e.ApplicationCommand
        Dim item As ToolStripItem = CType(e.Item, ToolStripItem)

        Select Case Command.ApplicationCommandName
            Case "New"
                e.ApplicationCommand.Enabled = _enableNewCheckBox.Checked
            Case "FileCopy"
                e.ApplicationCommand.Enabled = _enableGlobalCopyCheckBox.Checked
            Case "ContextCopy"
                e.ApplicationCommand.Enabled = _enableLocalCopyCheckBox.Checked
            Case "Save"
                e.ApplicationCommand.Visible = _showSaveCheckBox.Checked
            Case "TrackBar"
                _progressBar.Value = _trackBar.Value
        End Select
    End Sub

    Sub New()
        Me.InitializeComponent()
        Me.SetStyle(ControlStyles.Opaque, True)
    End Sub
End Class
